core/src/main/java/io/opencensus/stats/MeasureMap.java

```diff
@@ -0,0 +1,127 @@
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.stats;
+
+import io.opencensus.stats.Measure.DoubleMeasure;
+import io.opencensus.stats.Measure.LongMeasure;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * A map from {@link Measure}'s to measured values.
+ */
+public final class MeasureMap {
+
+  /**
+   * Returns a {@link Builder} for the {@link MeasureMap} class.
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Returns an {@link Iterator} over the measure/value mappings in this {@link MeasureMap}.
+   * The {@code Iterator} does not support {@link Iterator#remove()}.
+   */
+  Iterator<Measurement> iterator() {
+    return new MeasureMapIterator();
+  }
+
+  private final ArrayList<Measurement> measurements;
+
+  private MeasureMap(ArrayList<Measurement> measurements) {
+    this.measurements = measurements;
+  }
+
+  /**
+   * Builder for the {@link MeasureMap} class.
+   */
+  public static class Builder {
+    /**
+     * Associates the {@link DoubleMeasure} with the given value. Subsequent updates to the
+     * same {@link DoubleMeasure} are ignored.
+     *
+     * @param measure the {@link DoubleMeasure}
+     * @param value the value to be associated with {@code measure}
+     * @return this
+     */
+    public Builder put(DoubleMeasure measure, double value) {
+      measurements.add(Measurement.DoubleMeasurement.create(measure, value));
+      return this;
+    }
+
+    /**
+     * Associates the {@link LongMeasure} with the given value. Subsequent updates to the
+     * same {@link LongMeasure} are ignored.
+     *
+     * @param measure the {@link LongMeasure}
+     * @param value the value to be associated with {@code measure}
+     * @return this
+     */
+    public Builder put(LongMeasure measure, long value) {
+      measurements.add(Measurement.LongMeasurement.create(measure, value));
+      return this;
+    }
+
+    /**
+     * Constructs a {@link MeasureMap} from the current measurements.
+     */
+    public MeasureMap build() {
+      // Note: this makes adding measurements quadratic but is fastest for the sizes of
+      // MeasureMaps that we should see. We may want to go to a strategy of sort/eliminate
+      // for larger MeasureMaps.
+      for (int i = 0; i < measurements.size(); i++) {
+        String current =
+            measurements.get(i).getMeasure().getName();
+        for (int j = i + 1; j < measurements.size(); j++) {
+          if (current.equals(measurements.get(j).getMeasure().getName())) {
```

Comment:
I believe Measure's now overload equality (they originally did not) so you can simply use the equality operator here.